### ISSUE TITLE
QA Suite - pre-release trigger

### ISSUE THREAD

**Issue by ldraney** [department-of-veterans-affairs/notification-api#1327]
## User Story - Business Need
The business need for this ticket is to automatically trigger and run a full test suite against the staging environment upon a pre-release of the notification-api, ensuring code reliability before it reaches production.

### User Story(ies)
**As a** developer  
**I want** the full suite to be triggered from a pre-release, and only run against staging  
**So that** I can ensure the code is behaving as expected before it gets to production 

### Additional Info and Resources
- .github/workflows/prerelease_trigger.yml in the notification-api repo has only the pre-released triggers, and we only want the suite to be triggered from a pre-release, and to only run against staging.
- Again, ensure that the notification-api-qa repo's regression.yml only runs against staging upon this pre-release trigger

Based on #1442, this is the code to add at the end of the pre-release_trigger.yml file
```
  run-QA-suite:
    needs: [pre-release]
    uses: ./.github/workflows/regression-public.yml
    with:
      qa-image-branch: "master"
      environment: "staging"
    secrets: inherit
```

## Engineering Checklist
- [ ] Update [prerelease_trigger.yml](https://github.com/department-of-veterans-affairs/notification-api/blob/master/.github/workflows/prerelease_trigger.yml) to only trigger the suite from a pre-release, and to only run against staging, following the method determined in #1326
- [ ] Ensure the `regression.yml` is running correctly against the staging environment.

## Acceptance Criteria
- When a pre-release is triggered, the full suite should run against the staging environment.
- The `regression.yml` should run correctly against the staging environment.

## QA Considerations
- QA will need to verify that the full suite runs correctly against the staging environment when a pre-release is triggered.
- QA will need to validate that the `regression.yml` runs correctly against the staging environment.

## Potential Dependencies
- This ticket is dependent on [#1326](https://app.zenhub.com/workspaces/va-notify-620d21369d810a00146ed9c8/issues/gh/department-of-veterans-affairs/notification-api/1326). The developer will determine the best way to have repo to repo communication. Whether the GH CLI or deploy-keys were used, the same steps should be followed.
- We need to ensure the actions are completely solid in the QA repo before trying to trigger those actions for the notification-api repo.

## Out of Scope
- Running the suite for releases, this should only happen on pre-release.
- Running the suite against other environments than staging.

**Comment by mjones-oddball** [comment]
Note: Lucas felt this had less complexity and could be a 3 but we went with a 5 to be on the safe side. We should measure and be aware of the end result - did it get completed in less than a 5?

**Comment by cris-oddball** [comment]
@mjones-oddball based on the work in #1442, this ticket can be re-pointed to a 1 (and possibly brought into this sprint).

**Comment by mjones-oddball** [comment]
@cris-oddball if it's something you feel you have time to complete in the sprint, I'm on board. Otherwise I'll leave it as a stretch goal.

**Comment by cris-oddball** [comment]
Code approved and merged.
Tested first by cutting and pushing a tag to Perf. The regression ran in Perf as expected.
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBN1VaQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--fd9111d7ebaa9f767f1bcdb26cde6d9cc11431ae/Screen%20Shot%202023-10-09%20at%2011.09.20%20AM.jpg)

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBN1laQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--bb969edec53faca6aa01ee21ffffbc78970c6183/Screen%20Shot%202023-10-09%20at%2011.08.45%20AM.jpg)

Next, creating a pre-release from this tag and sending it up to Staging.

**Comment by cris-oddball** [comment]
[Pre-release to staging](https://github.com/department-of-veterans-affairs/notification-api/actions/runs/6459730710).

**Comment by cris-oddball** [comment]
The pre-release did run the regression against staging, but the regression also failed on three tests.

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOG9aQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--288f4a995815173a34339ef6bbde98bdaebef222/Screen%20Shot%202023-10-09%20at%2011.28.27%20AM.jpg)


[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOHNaQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--b35d7aea326d8dc4371cf5cf10ae551ca1022329/Screen%20Shot%202023-10-09%20at%2011.25.35%20AM.jpg)

Details of test failures showing connection timeouts to the public endpoints:
```
FAILED tests/test_communication_item.py::test_update_communication_item_negative_tests_400[Payload Name empty string]
E       AssertionError: Invalid Response: Requests Exception! failing test: HTTPSConnectionPool(host='staging-api.va.gov', port=443): Max retries exceeded with url: /vanotify/communication-item/55bf920f-681e-4356-987b-776b1efb1c2d (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x7fd9cdae2260>, 'Connection to staging-api.va.gov timed out. (connect timeout=3.05)'))

FAILED tests/test_communication_item.py::test_get_communication_item_invalid_communication_item_404
E       AssertionError: Invalid Response: Requests Exception! failing test: HTTPSConnectionPool(host='staging-api.va.gov', port=443): Max retries exceeded with url: /vanotify/communication-item/9bec2fbf-c7cc-4ade-97cf-c5ccac874bfb (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x7fd9cb3c3100>, 'Connection to staging-api.va.gov timed out. (connect timeout=3.05)'))

FAILED tests/test_service_callback.py::test_delete_callback_webhook - Asserti...
E       AssertionError: Invalid Response: Requests Exception! failing test: HTTPSConnectionPool(host='staging-api.va.gov', port=443): Max retries exceeded with url: /vanotify/service/d73db30a-2883-478f-92bd-d30fee8b4282/callback/d89ed516-f985-44db-84a2-02ec75db8817 (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x7fd9cb3c3ac0>, 'Connection to staging-api.va.gov timed out. (connect timeout=3.05)'))

FAILED tests/test_v2_notifications_email.py::test_send_email[Email: Basic Formatting w/ BIRLSID and Optional Properties]
E       AssertionError: Invalid Response: Requests Exception! failing test: HTTPSConnectionPool(host='staging-api.va.gov', port=443): Max retries exceeded with url: /vanotify/v2/notifications/email (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x7fd9ce379a80>, 'Connection to staging-api.va.gov timed out. (connect timeout=3.05)'))
```

Rerunning the failed tests.

**Comment by cris-oddball** [comment]
3rd time is the charm, but will need to see if this impacts staging pre-releases. Insert a time delay?
[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBL2daQVE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--efe4a003b6bae1d5cafae0a9d04e60f09dfaa229/Screen%20Shot%202023-10-09%20at%2011.55.36%20AM.jpg)

**Comment by cris-oddball** [comment]
Opened a new PR to add a 10 second delay between the deploy completing and the regression running. Awaiting approvals.

**Comment by cris-oddball** [comment]
Draft PR on hold since Kyle wanted to investigate.

**Comment by cris-oddball** [comment]
Reverted PR by means of a new PR. Removed run-qa-suite from all API workflows. Closing this as Will Not Do since we now need a new plan.
